azure-pipelines.yml

```diff
@@ -57,6 +61,23 @@ jobs:
       with_toxenv: 'coverage' # generate .tox/.coverage, .tox/coverage.xml after test run
       for_envs: [py37, py36, py35, py34, py27]
 
+- job: py39_linux
```

Comment:
This should not be done here like this; instead, let's pull python 3.9 in via deadsnake at https://github.com/tox-dev/azure-pipelines-template/pull/3/commits/a7025f61b619028dfecbbb2c0889f7e580a9db02

Reply:
please add the pr to tox-dev/azure-pipelines first please 👍

Reply:
Ah, I missed that!

It looks like the deadsnakes py39 is [broken the same way](https://dev.azure.com/pypa/virtualenv/_build/results?buildId=15624), though -- it vendors a copy of `pkg_resources` that tries to work with `collections.MutableMapping`.

Reply:
And, nod, I figured this wasn't the right way to go about it, and would need to be integrated into the tox-dev template matrix, but I wanted to stand up a 3.9 job ASAP.

Reply:
> please add the pr to tox-dev/azure-pipelines first please 👍

Not sure what you mean by this... should I just wait on this until that branch of tox-dev/azure-pipelines-template is merged?

Reply:
that would be the merge requirement, but just to test it's fine to add it inline for now; @asottile any ideas about deadsnakes py39 being broken?

Reply:
@gaborbernat Is it possible to point Azure at a commit? This error makes me think that it only likes branches:

> `/azure-pipelines.yml: Could not get the latest source version for repository tox-dev/azure-pipelines-template hosted on http://github.com/ using ref refs/heads/commit/a7025f6. GitHub reported the error, "No commit found for SHA: refs/heads/commit/a7025f6"`

Can you revert your ["no 3.9 yet" commit](https://github.com/laike9m/azure-pipelines-template/commit/df0dbec1d3e5ff3540e114b2f54aee4c8b2e47ed) on the template fork?

Reply:
please just fill it in again 👍

Reply:
is there a trace to look at? I only just uploaded the packages last night

Reply:
@gaborbernat How do I do that? I don't have write bit on the forked repo.

@asottile See the `py39_linux` job at https://dev.azure.com/pypa/virtualenv/_build/results?buildId=15624

Reply:
that looks like old pip - probably the debian upstream one -- try get-pip.py? (or make a virtualenv)

Reply:
@asottile Some sort of problem on install of `python3.9-dev`: https://dev.azure.com/pypa/virtualenv/_build/results?buildId=15629

Reply:
nah just yaml is hard :P you've got `displayName:` indented too much

Reply:
Lol, yup, realized that 2sec after I pinged you. Sry.

Reply:
(that warning that shows up is annoying though, but I haven't figured out how to fix it quite yet -- low priority though since it appears to be harmless)

Reply:
@gaborbernat @asottile No idea why I'm getting a permissions error here: https://dev.azure.com/pypa/virtualenv/_build/results?buildId=15642

Reply:
some bug with tox's isolated builds? hard to tell -- might need to debug a bit -- have you been able to reproduce it locally?

Reply:
Haven't seriously tried yet -- my actual code change lets me create a virtualenv just fine now, locally; but, I haven't yet tried to run tox CI on 3.9. I'm frankly somewhat baffled as to what the correct sequence of steps is.

Reply:
Hah, really? just `pip install -e . && tox -e py39`? Boy, did I overthink *that*.